src/controllers/searchController.ts

```diff
@@ -1,6 +1,7 @@
 import { Request, Response, NextFunction } from 'express';
+import Category from '../models/Category';
 import Stall from '../models/Stall';
-import { findStallsByIdsFunc, mapStallToCardFunc } from './stallController';
+import { findAllStallsFunc, findStallsByIdsFunc } from './stallController';
 
 async function searchStalls(req: Request, res: Response, next: NextFunction) {
   try {
```

Comment:
If `req.params.query` is empty, the query won't return any values? If it doesn't, just manually handle the case here.

Reply:
If the `req.query.param` is empty, it will return all stalls.

Reply:
That means your reduce function will return all ids right? -> should be able to go through normal middleware chain for `search`